### PR TITLE
feat: TokenSelectorDropdown

### PR DESCRIPTION
**Pull request by kyhyco**
**What changed? Why?**
- add `TokenSelectorDropdown` to be used with `TokenSelectorDropdown`

Screenshot 2024-06-07 at 10 43 55 AM

With scrollbar
Screenshot 2024-06-07 at 10 42 00 AM

**Notes to reviewers**
`TokenSelector` and `TokenSelectorDropdown` should be built with non-div elements like `select` and `option`. I will address this in future PRs.

**How has it been tested?**
locally

### CODE REVIEW

File: src/token/components/TokenSelectorDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { useCallback, useEffect, useRef } from 'react';
+import { TokenSelectorDropdownReact } from '../types';
+import { TokenRow } from './TokenRow';
+
+export function TokenSelectorDropdown({ setToken, options, onToggle }: TokenSelectorDropdownReact) {
+  const ref = useRef<HTMLDivElement>(null);
+
+  const handleBlur = useCallback((event: MouseEvent) => {
+    /* istanbul ignore next */
```
Comment (kyhyco) [inline src/token/components/TokenSelectorDropdown.tsx:9]: this part is tested but not sure why coverage report does not pick this up

File: site/docs/pages/token/token-selector.mdx [hunk 2]
```diff
@@ -1,107 +1,129 @@
...
-      symbol: 'ETH',
-    }}
-    onClick={() => {}}
-  />
-</App>
-
-`TokenSelector` with no token prop
-
-:::code-group
+## Props
```
Comment (kyhyco) [inline site/docs/pages/token/token-selector.mdx:88]: Had to remove the code here. This needs to be put back after a new release has been cut

```
<App>
  <TokenSelectorContainer>
    {(token, setToken) => (
      <TokenSelector token={token}>
        <TokenSelectorDropdown
          setToken={setToken}
          options={[
            {
              name: 'Ethereum',
              address: '',
              symbol: 'ETH',
              decimals: 18,
              image: 'https://wallet-api-production.s3.amazonaws.com/uploads/tokens/eth_288.png',
              blockchain: 'eth',
              chainId: 8453,
            },
            {
              name: 'USDC',
              address: '0x833589fcd6edb6e08f4c7c32d4f71b54bda02913',
              symbol: 'USDC',
              decimals: 6,
              image:
                'https://d3r81g40ycuhqg.cloudfront.net/wallet/wais/44/2b/442b80bd16af0c0d9b22e03a16753823fe826e5bfd457292b55fa0ba8c1ba213-ZWUzYjJmZGUtMDYxNy00NDcyLTg0NjQtMWI4OGEwYjBiODE2',
              blockchain: 'eth',
              chainId: 8453,
            },
            {
              name: 'Dai',
              address: '0x50c5725949a6f0c72e6c4a641f24049a917db0cb',
              symbol: 'DAI',
              decimals: 18,
              image:
                'https://d3r81g40ycuhqg.cloudfront.net/wallet/wais/d0/d7/d0d7784975771dbbac9a22c8c0c12928cc6f658cbcf2bbbf7c909f0fa2426dec-NmU4ZWViMDItOTQyYy00Yjk5LTkzODUtNGJlZmJiMTUxOTgy',
              blockchain: 'eth',
              chainId: 8453,
            },
          ]}
        />
      </TokenSelector>
    )}
  </TokenSelectorContainer>
</App>
```

File: site/docs/pages/token/token-selector.mdx [hunk 3]
```diff
@@ -1,107 +1,129 @@
-import { TokenSelector } from '@coinbase/onchainkit/token';
+{/* import { TokenSelector, TokenSelectorDropdown } from '../../../../src/token'; */}
```
Comment (kyhyco) [inline site/docs/pages/token/token-selector.mdx:1]: Uncomment this after new release

File: site/docs/components/TokenSelectorContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { ReactElement, useState } from 'react';
+import type { Token } from '@coinbase/onchainkit/token';
+
+type TokenSelectorContainer = {
```
Comment (Zizzamia) [inline site/docs/components/TokenSelectorContainer.tsx:4]: btw let's keep have all the types in the types.ts file, even the one we don't export.

File: site/docs/pages/token/types.mdx [hunk 5]
```diff
@@ -93,7 +93,18 @@ export type TokenSearchReact = {
 
 ```ts
 export type TokenSelectorReact = {
-  token: Token; // Rendered token
-  onClick: () => void; // Component on click handler
+  children: ReactElement<{ setToken: (token: Token) => void; onToggle: () => void }>; // Should either be a dropdown or modal that handle token selection
```
Comment (Zizzamia) [inline site/docs/pages/token/types.mdx:96]: Should this type `{ setToken: (token: Token) => void; onToggle: () => void }` being express in a more clean way? Maybe his own type?

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**